ov/appliance_snmp_v3_trap_destinations.go

```diff
@@ -0,0 +1,231 @@
+	NextPageURI utils.Nstring `json:"nextPageUri,omitempty"` // "nextPageUri": null,
+	PrevPageURI utils.Nstring `json:"prevPageUri,omitempty"` // "prevPageUri": null,
+	Start       int           `json:"start,omitempty"`       // "start": 0,
+	Total       int           `json:"total,omitempty"`       // "total": 1,
+	Type        string        `json:"type,omitempty"`
+	URI         utils.Nstring `json:"uri,omitempty"`
+}
+
+type ValidateSNMPv3Address struct {
+	DestinationAddress   string          `json:"destinationAddress,omitempty"`
+	ExistingDestinations []utils.Nstring `json:"existingDestinations,omitempty"`
+}
+
+func (c *OVClient) ValidateDestinationAddress(destId string, existId []utils.Nstring) error {
+	var (
+		uri = "/rest/appliance/snmpv3-trap-forwarding/destinations/validation"
+	)
+
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	snmpUser2 := ValidateSNMPv3Address{
+		DestinationAddress:   destId,
+		ExistingDestinations: existId,
+	}
+	log.Debugf("REST : %s \n %+v\n", uri, destId)
+	_, err := c.RestAPICall(rest.POST, uri, snmpUser2)
+	if err != nil {
+		log.Errorf("Error submitting validating destination address request: %s", err)
+		return err
+	}
+
+	return nil
+}
+
+func (c *OVClient) CreateSNMPv3TrapDestinations(trapOption SNMPv3Trap) error {
+	log.Infof("Initializing creation of SNMPv3 Trap Destinations for %s.", trapOption.UserID)
+	var (
+		uri      = "/rest/appliance/snmpv3-trap-forwarding/destinations"
+		trapdata SNMPv3Trap
+	)
+	//validating the Destination Address
+	log.Infof("Validating SNMPv3 Trap Destinations Address %s.", trapOption.DestinationAddress)
+	err := c.ValidateDestinationAddress(trapOption.DestinationAddress, *new([]utils.Nstring))
+	if err != nil {
+		return errors.New("Invalid Destination Address")
+	}
+	log.Infof("Successfully validated the Destination Address.")
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	log.Debugf("REST : %s \n %+v\n", uri, trapOption)
+	data, err := c.RestAPICall(rest.POST, uri, trapOption)
+	if err != nil {
+		log.Errorf("Error submitting new snmpv3 trap destinations request: %s", err)
+		return err
+	}
+
+	log.Debugf("Response New SNMPv3 Trap Destinations %s", data)
+	if err := json.Unmarshal(data, &trapdata); err != nil {
+		log.Errorf("Error with task un-marshal: %s", err)
+		return err
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *OVClient) GetSNMPv3TrapDestinations(filter string, sort string, start string, count string) (SNMPv3TrapList, error) {
+	var (
+		uri      = "/rest/appliance/snmpv3-trap-forwarding/destinations"
+		q        = make(map[string]interface{})
+		traplist SNMPv3TrapList
+	)
+
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	if len(filter) > 0 {
+		q["filter"] = filter
+	}
+
+	if sort != "" {
+		q["sort"] = sort
+	}
+
+	if start != "" {
+		q["start"] = start
+	}
+
+	if count != "" {
+		q["count"] = count
+	}
+	if len(q) > 0 {
+		c.SetQueryString(q)
```

Comment:
Since you are making changes to ```RestAPICall``` method in one of your as part of issue #315 , why don't you follow the same approach for all the remaining resources..?
```data, err := c.RestAPICall(rest.GET, uri, nil, q)```